python/ray/serve/common.py

```diff
@@ -49,24 +48,28 @@ class ReplicaName:
     replica_suffix: str
     replica_tag: ReplicaTag = ""
     delimiter: str = "#"
+    prefix: str = "SERVE_REPLICA::"
 
     def __init__(self, deployment_tag: str, replica_suffix: str):
         self.deployment_tag = deployment_tag
         self.replica_suffix = replica_suffix
         self.replica_tag = f"{deployment_tag}{self.delimiter}{replica_suffix}"
 
+    @staticmethod
+    def is_replica_name(actor_name: str) -> bool:
+        return (actor_name.startswith(ReplicaName.prefix)
+                and actor_name.count("#") == 1)
+
     @classmethod
-    def from_str(self, replica_name):
-        parsed = replica_name.split(self.delimiter)
+    def from_str(cls, actor_name):
+        assert actor_name.startswith(cls.prefix), "Actor "
+        replica_name = actor_name.replace(cls.prefix, "")
+
+        parsed = replica_name.split(cls.delimiter)
         assert len(parsed) == 2, (
             f"Given replica name {replica_name} didn't match pattern, please "
-            f"ensure it has exactly two fields with delimiter {self.delimiter}"
-        )
-        self.deployment_tag = parsed[0]
-        self.replica_suffix = parsed[1]
-        self.replica_tag = replica_name
-
-        return self
+            f"ensure it has exactly two fields with delimiter {cls.delimiter}")
+        return cls(deployment_tag=parsed[0], replica_suffix=parsed[1])
```

Comment:
this never worked before. seems like the old way is overriding the class attribute directly and its state lingers, might have cause some weird bug before